src/dsync/dsync.c

```diff
@@ -2968,6 +2971,20 @@ static int dsync_daos(
 }
 #endif
 
+static bool is_mtime_nsec_used(char *srcpath, char *dstpath)
```

Comment:
(nit) If you refresh the patch, it would be good to add a comment here that this is a heuristic - we're guessing, that if tv_nsec == 0, that the file system doesn't support nsec even though 0 is a valid value and sometimes our check might be wrong.

No need to re-do just for this though, since that's described in the issue.